sdk/rust/oak/src/lib.rs

```diff
@@ -401,3 +401,43 @@ pub fn set_panic_hook() {
         );
     }));
 }
+
+/// Trait implemented by all Oak Nodes.
+///
+/// It has a single method for handling commands, which are [`Decodable`](crate::io::Decodable)
+/// objects that are received via the single incoming channel handle which is passed in at node
+/// creation time.
+pub trait Node<T: crate::io::Decodable> {
+    fn handle_command(&mut self, command: T) -> Result<(), crate::OakError>;
+}
+
+pub fn run_event_loop<T: crate::io::Decodable, N: Node<T>>(mut node: N, in_handle: u64) {
+    let in_channel = crate::ReadHandle {
+        handle: crate::Handle::from_raw(in_handle),
+    };
+    if !in_channel.handle.is_valid() {
+        error!("invalid input handle");
+        return;
+    }
+    let receiver = crate::io::Receiver::new(in_channel);
+    info!("starting event loop");
+    loop {
+        // First wait until a message is available. If the node was terminated while waiting, this
+        // will return `ERR_TERMINATED`, which indicates that the event loop should be terminated.
+        if let Err(crate::OakStatus::ERR_TERMINATED) = receiver.wait() {
+            info!("terminating event loop");
+            return;
+        }
```

Comment:
Do we need to handle other errors here?

Reply:
Made it a bit more explicit, not sure if there's anything else we can do, let me know if you have any ideas.

Reply:
LGTM; the only thing I wonder is whether another error should terminate the loop.  E.g. we got into a state where the underlying handle had been closed somehow (not sure how, I don't think it's possible), then would this infinite-loop with `ERR_BAD_HANDLE`?

Reply:
Added more handling logic and expanded comment, PTAL.